requirements/extras/test_requirements.txt

```diff
@@ -39,3 +39,4 @@ tritonclient[http]<2.37.0
 onnx==1.14.1
 # tf2onnx==1.15.1
 nbformat>=5.9,<6
+accelerate>=0.24.1,<=0.27.0
```

Comment:
1. We need this dependency to calculate model size. This is also the preferred approach for adding new dependencies. I think installing it on the fly may not work, since the functionality to calculate model size was added after v0.23.0.

https://github.com/huggingface/accelerate/releases/tag/v0.23.0 

2. There are quite a few dependency that are shared between the deployment types. And this clean up should ideally be a dedicated task to avoid mistakes.

Reply:
I understand the problem, but I am not in favor of the solution (like bringing new external dependency) because of the reasons stated in the PR above. Even today, I received an external PR (https://github.com/aws/sagemaker-python-sdk/pull/4452). The point I want to make is, if the dependency is for a specific case, then I suggest coming up with an alternative. If it is used across different components, then it is easy to maintain. 

There is also a security issue with another dependency: https://github.com/aws/sagemaker-python-sdk/security/dependabot/53

Reply:
There doesn't seem to be an alternative here for now in terms of using estimate.parser. So if there are any ideas please me know.

Reply:
Also, I've added the dependency to extras, so it should not be a mandatory dependency to install for customers. Is this understanding correct?

Reply:
Update: Discussed offline, move estimate_parser code utilities to be in the SDK (with some risks) to prevent all customers from installing accelerate.